apps/recovery-portal/src/main/webapp/self-registration-complete.jsp

```diff
@@ -37,6 +51,40 @@
 
     isEmailNotificationEnabled = Boolean.parseBoolean(application.getInitParameter(
             IdentityManagementEndpointConstants.ConfigConstants.ENABLE_EMAIL_NOTIFICATION));
+
+    if (tenantDomain != null) {
+        isAutoLoginEnable = Boolean.parseBoolean(Utils.getConnectorConfig("SelfRegistration.AutoLogin.Enable",
+                tenantDomain));
+        username = username + "@" + tenantDomain;
+    }
+    if (userStoreDomain != null) {
+        username = userStoreDomain + "/" + username;
+    }
+    if (isAutoLoginEnable) {
+        String queryParams = callback.substring(callback.indexOf("?") + 1);
+        String[] parameterList = queryParams.split("&");
+        Map<String, String> queryMap = new HashMap<>();
+        for (String param : parameterList) {
+            String key = param.substring(0, param.indexOf("="));
+            String value = param.substring(param.indexOf("=") + 1);
+            queryMap.put(key, value);
+        }
+        sessionDataKey = queryMap.get("sessionDataKey");
+        if (userStoreDomain != null) {
+          username = userStoreDomain + "/" + username;
+        }
+        String signature = Base64.getEncoder().encodeToString(SignatureUtil.doSignature(username));
+        JSONObject cookieValueInJson = new JSONObject();
+        cookieValueInJson.put("username", username);
+        cookieValueInJson.put("signature", signature);
```

Comment:
Looks like this is not the page where we validate the confirmation code. Need to set the cookie in the same place we validate the code. otherwise, if someone directly comes to this page, he can login automatically.

Reply:
fixed in [3c7c5de](https://github.com/wso2/identity-apps/pull/1409/commits/3c7c5dea9f37773bb94226823631aa38b36f810c)